File: CS/CS/WindowsApplication1/CustomEditor/GetButtonCaptionEventArgs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraEditors.Controls;

namespace WindowsApplication1.CustomEditor
{
    public delegate void GetButtonCaptionDelegate(ButtonEditCustomCaptionHelper helper, GetButtonCaptionEventArgs args);

    public class GetButtonCaptionEventArgs : EventArgs
    {

        // Fields...
        private string _ButtonText;
        private EditorButton _Button;
        private GridColumn _Column;
        private int _RowHandle;
        private GridView _View;

        /// <summary>
        /// Initializes a new instance of the GetButtonCaptionEventArgs class.
        /// </summary>
        /// <param name="button"></param>
        /// <param name="column"></param>
        /// <param name="rowHandle"></param>
        /// <param name="view"></param>
        public GetButtonCaptionEventArgs(EditorButton button, GridColumn column, int rowHandle, GridView view)
        {
            _Button = button;
            _Column = column;
            _RowHandle = rowHandle;
            _View = view;
            _ButtonText = button.Caption;
        }

        public GridView View
        {
            get { return _View; }
            set
            {
                _View = value;

            }
        }


        public int RowHandle
        {
            get { return _RowHandle; }
            set
            {
                _RowHandle = value;

            }
        }

        public GridColumn Column
        {
            get { return _Column; }
            set
            {
                _Column = value;

            }
        }


        public EditorButton Button
        {
            get { return _Button; }
            set
            {
                _Button = value;

            }
        }



        public string ButtonText
        {
            get { return _ButtonText; }
            set
            {
                _ButtonText = value;

            }
        }
    }
}

File: VB/WindowsApplication1/CustomEditor/GetButtonCaptionEventArgs.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraEditors.Controls

Namespace WindowsApplication1.CustomEditor
	Public Delegate Sub GetButtonCaptionDelegate(ByVal helper As ButtonEditCustomCaptionHelper, ByVal args As GetButtonCaptionEventArgs)

	Public Class GetButtonCaptionEventArgs
		Inherits EventArgs

		' Fields...
		Private _ButtonText As String
		Private _Button As EditorButton
		Private _Column As GridColumn
		Private _RowHandle As Integer
		Private _View As GridView

		''' <summary>
		''' Initializes a new instance of the GetButtonCaptionEventArgs class.
		''' </summary>
		''' <param name="button"></param>
		''' <param name="column"></param>
		''' <param name="rowHandle"></param>
		''' <param name="view"></param>
		Public Sub New(ByVal button As EditorButton, ByVal column As GridColumn, ByVal rowHandle As Integer, ByVal view As GridView)
			_Button = button
			_Column = column
			_RowHandle = rowHandle
			_View = view
			_ButtonText = button.Caption
		End Sub

		Public Property View() As GridView
			Get
				Return _View
			End Get
			Set(ByVal value As GridView)
				_View = value

			End Set
		End Property


		Public Property RowHandle() As Integer
			Get
				Return _RowHandle
			End Get
			Set(ByVal value As Integer)
				_RowHandle = value

			End Set
		End Property

		Public Property Column() As GridColumn
			Get
				Return _Column
			End Get
			Set(ByVal value As GridColumn)
				_Column = value

			End Set
		End Property


		Public Property Button() As EditorButton
			Get
				Return _Button
			End Get
			Set(ByVal value As EditorButton)
				_Button = value

			End Set
		End Property



		Public Property ButtonText() As String
			Get
				Return _ButtonText
			End Get
			Set(ByVal value As String)
				_ButtonText = value

			End Set
		End Property
	End Class
End Namespace
